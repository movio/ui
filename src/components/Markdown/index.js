import React, { PropTypes } from 'react';
import ReactMarkdown from 'react-markdown';
import classnames from 'classnames';
import CodeRenderer from './CodeRenderer';

import styles from 'components/Markdown/markdown.css';

const Markdown = ({ source, className, overrideMarkdownStyle }) =>
  <div className={classnames(className, styles.container)}>
    <ReactMarkdown
      source={source}
      className={overrideMarkdownStyle ? '' : styles.markdown}
      escapeHtml
      renderers={{
        CodeBlock: CodeRenderer,
        Code: CodeRenderer,
      }}
    />
  </div>;

Markdown.propTypes = {
  source: PropTypes.string.isRequired,
  className: PropTypes.string,
  overrideMarkdownStyle: PropTypes.bool,
};

export default Markdown;

export { styles };
