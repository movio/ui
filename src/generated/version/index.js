// This file is generated

import * as getByOrgkeyAndApplicationkeyAndVersion
  from 'generated/version/getByOrgkeyAndApplicationkeyAndVersion';

const actionTypes = Object.assign(
  {},
  getByOrgkeyAndApplicationkeyAndVersion.actionTypes
);

const actions = Object.assign(
  {},
  getByOrgkeyAndApplicationkeyAndVersion.actions
);

const sagas = {
  getByOrgkeyAndApplicationkeyAndVersionTakeEverySaga: getByOrgkeyAndApplicationkeyAndVersion.takeEverySaga,
  getByOrgkeyAndApplicationkeyAndVersionLatestSaga: getByOrgkeyAndApplicationkeyAndVersion.takeLatestSaga,
};

const unloadAction = 'UNLOAD_APPLICATION';

export { actionTypes, actions, sagas, unloadAction };
