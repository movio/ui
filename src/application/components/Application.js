// @flow
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import LoadingOverlay from 'components/LoadingOverlay';
import Operation from 'application/components/Operation';
import ApplicationHome from 'application/components/ApplicationHome';
import * as utils from 'utils';
import Model from 'application/components/Model';
import Enum from 'application/components/Enum';

import type { Service } from 'generated/version/ServiceType';

import styles from 'application/components/application.css';

import { actions as serviceActions, unload_action } from 'generated/version';

const allActions = Object.assign({}, serviceActions);

type Params = {
  model?: string,
  resource?: string,
  method: string,
  path: string,
  applicationKey: string,
  organizationKey: string,
};

type Props = {
  actions: Object, // FIXME
  params: Params, // FIXME
  loaded: boolean,
  service: Service,
  importedServices: Service[],
};

export class Application extends Component {
  props: Props;

  componentDidMount() {
    const orgKey = this.props.params.organizationKey;
    const applicationKey = this.props.params.applicationKey;
    this.props.actions.getByOrgkeyAndApplicationkeyAndVersion_get({
      orgKey,
      applicationKey,
      version: 'latest',
    });
  }

  componentWillUnmount() {
    this.props.onUnload();
  }

  render() {
    const { service, importedServices, loaded } = this.props;

    if (!service || !loaded) {
      return <LoadingOverlay isLoaded={loaded} />;
    }
    if (this.props.params.resource) {
      // Load Operation
      const {
        method,
        path,
        resource: resourceType,
        applicationKey,
        organizationKey,
      } = this.props.params;

      const operation = utils.getOperation(resourceType, method, path, service);
      const resource = utils.getResource(resourceType, service);

      return (
        <LoadingOverlay isLoaded={loaded}>
          <Operation
            service={service}
            importedServices={importedServices}
            operation={operation}
            applicationKey={applicationKey}
            organizationKey={organizationKey}
            resource={resource}
            method={method}
            path={path}
          />
        </LoadingOverlay>
      );
    } else if (this.props.params.model) {
      // Load Model
      const modelName = this.props.params.model;
      if (utils.isEnum(modelName, service, importedServices)) {
        const enumModel = utils.getEnum(modelName, service, importedServices);
        return enumModel
          ? <LoadingOverlay isLoaded={loaded}>
              <Enum
                enumModel={enumModel}
                service={service}
                importedServices={importedServices}
              />
            </LoadingOverlay>
          : null;
      } else {
        const model = utils.getModel(modelName, service, importedServices);
        return model
          ? <LoadingOverlay isLoaded={loaded}>
              <Model
                model={model}
                service={service}
                importedServices={importedServices}
                showJsonDoc={true}
              />
            </LoadingOverlay>
          : null;
      }
    } else {
      // Load Application Home
      const { applicationKey, organizationKey } = this.props.params;

      return (
        <LoadingOverlay isLoaded={loaded}>
          <ApplicationHome
            service={service}
            applicationKey={applicationKey}
            organizationKey={organizationKey}
          />
        </LoadingOverlay>
      );
    }
  }
}

const mapStateToProps = state => ({
  loaded: state.application.loaded,
  service: state.application.service,
  importedServices: state.application.importedServices,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(allActions, dispatch),
  onUnload: () => dispatch({ type: unload_action }),
});

export default connect(mapStateToProps, mapDispatchToProps)(Application);

export { styles };
