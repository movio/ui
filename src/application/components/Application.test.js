// @flow
import React from 'react';
import renderer from 'react-test-renderer';

import { Application } from 'application/components/Application';
import Example from 'exampleService.json';
import { actions as serviceActions } from 'generated/version';

test('Application Home', () => {
  const params = {
    actions: serviceActions,
    onUnload:  () => {},
    params: {
      model: undefined,
      resource: undefined,
      method: '',
      path: '',
      applicationKey: '',
      organizationKey: '',
    },
    loaded: true,
    service: Example,
    importedServices: [],
  };

  const component = renderer.create(<Application {...params} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
