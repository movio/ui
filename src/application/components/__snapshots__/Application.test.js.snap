// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Home 1`] = `
<div>
  <div>
    <div
      className="content"
    >
      <div
        className="header"
      >
        <h1
          className="h1 h1"
        >
          Apidoc Generator
        </h1>
        <div
          className="description container"
        >
          <div
            className="markdown"
          >
            <p>
              Documentation for an apidoc code generator API
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          className="container isClick"
        >
          <a
            onClick={[Function]}
            tabIndex="0"
          >
            <div
              className="flex"
            >
              <div
                className="left"
              >
                <div
                  className="method get isExpandedMethod"
                >
                  GET
                </div>
              </div>
              <div
                className="right"
              >
                <pre
                  className="path isExpandedPath"
                >
                  /generators
                </pre>
              </div>
            </div>
            <div
              className="description container"
            >
              <div
                className="markdown"
              >
                <p>
                  Get all available generators
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          className="container isClick"
        >
          <a
            onClick={[Function]}
            tabIndex="0"
          >
            <div
              className="flex"
            >
              <div
                className="left"
              >
                <div
                  className="method get isExpandedMethod"
                >
                  GET
                </div>
              </div>
              <div
                className="right"
              >
                <pre
                  className="path isExpandedPath"
                >
                  /generators/:key
                </pre>
              </div>
            </div>
            <div
              className="description container"
            >
              <div
                className="markdown"
              >
                <p>
                  Get generator with this key
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          className="container isClick"
        >
          <a
            onClick={[Function]}
            tabIndex="0"
          >
            <div
              className="flex"
            >
              <div
                className="left"
              >
                <div
                  className="method get"
                >
                  GET
                </div>
              </div>
              <div
                className="right"
              >
                <pre
                  className="path"
                >
                  /_internal_/healthcheck
                </pre>
              </div>
            </div>
          </a>
        </div>
        <div
          className="container isClick"
        >
          <a
            onClick={[Function]}
            tabIndex="0"
          >
            <div
              className="flex"
            >
              <div
                className="left"
              >
                <div
                  className="method post isExpandedMethod"
                >
                  POST
                </div>
              </div>
              <div
                className="right"
              >
                <pre
                  className="path isExpandedPath"
                >
                  /invocations/:key
                </pre>
              </div>
            </div>
            <div
              className="description container"
            >
              <div
                className="markdown"
              >
                <p>
                  Invoke a generator
                </p>
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
